### ISSUE TITLE
Failed to infer rainbow table parameters

### ISSUE THREAD

**Issue by GeedieZ** [jtesta/rainbowcrackalack#7]
Trying to run your program under newest Windows 10 machine. Downloaded the 3,5 GB torrent and checked the MD5SUM - all correct. But When I try to run the app it says something like this:

```
PS D:\hg\Rainbow Crackalack v1.0> .\crackalack_lookup.exe .\rt_ntlm\  D:\hg\Hashes\nthashes.txt

Rainbow Crackalack v1.0
Copyright 2018-2019 Positron Security LLC <https://www.positronsecurity.com/>
Make Rainbow Tables Great Again


Found 2 platforms.
Found 1 devices on platform #0.
Found 2 devices on platform #1.
Device #0:
        Vendor: NVIDIA Corporation
        Name: GeForce GTX 1060
        Version: OpenCL 1.2 CUDA
        Driver: 430.86
        Max compute units: 10
        Max work group size: 1024
        Global memory size: 3221225472

Device #1:
        Vendor: Intel(R) Corporation
        Name: Intel(R) HD Graphics 530
        Version: OpenCL 2.0
        Driver: 22.20.16.4749
        Max compute units: 24
        Max work group size: 256
        Global memory size: 3374045594

Binary searching will be done with 8 threads.
Loaded 44881 of 44881 uncracked hashes from D:\hg\Hashes\nthashes.txt. Failed to infer rainbow table parameters from files in directory.  Ensure that valid rainbow table files are in .\rt_ntlm\ (and/or its sub-directories).
PS D:\hg\Rainbow Crackalack v1.0>
```

File in folder:
PS D:\hg\Rainbow Crackalack v1.0\rt_ntlm> ls

    Directory: D:\hg\Rainbow Crackalack v1.0\rt_ntlm

```

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       13.06.2019     19:44           1090 ntlm_loweralpha-numeric-space#1-8_0.md5sums
-a----       13.06.2019     19:56      152302846 ntlm_loweralpha-numeric-space#1-8_0_10000x24663209_distrrtgen[p][i]_9.rti2
-a----       13.06.2019     19:57      414413957 ntlm_loweralpha-numeric-space#1-8_0_10000x67108864_distrrtgen[p][i]_0.rti2
-a----       13.06.2019     19:57      414413911 ntlm_loweralpha-numeric-space#1-8_0_10000x67108864_distrrtgen[p][i]_1.rti2
-a----       13.06.2019     19:57      414412536 ntlm_loweralpha-numeric-space#1-8_0_10000x67108864_distrrtgen[p][i]_2.rti2
-a----       13.06.2019     19:57      414414591 ntlm_loweralpha-numeric-space#1-8_0_10000x67108864_distrrtgen[p][i]_3.rti2
-a----       13.06.2019     19:56      414414823 ntlm_loweralpha-numeric-space#1-8_0_10000x67108864_distrrtgen[p][i]_4.rti2
-a----       13.06.2019     19:57      414414054 ntlm_loweralpha-numeric-space#1-8_0_10000x67108864_distrrtgen[p][i]_5.rti2
-a----       13.06.2019     19:57      414414561 ntlm_loweralpha-numeric-space#1-8_0_10000x67108864_distrrtgen[p][i]_6.rti2
-a----       13.06.2019     19:57      414414181 ntlm_loweralpha-numeric-space#1-8_0_10000x67108864_distrrtgen[p][i]_7.rti2
-a----       13.06.2019     19:57      414416139 ntlm_loweralpha-numeric-space#1-8_0_10000x67108864_distrrtgen[p][i]_8.rti2
```

**Comment by jtesta** [comment]
Try giving an absolute path to the rt_ntlm directory, instead of a relative one.  Or perhaps even try "rt_ntlm" on its own, and not include the dot-slash prefix?

Let me know what works, if anything.  If its not liking relative paths, then this is a bug.

**Comment by GeedieZ** [comment]
Sorry, still no dice
Exactly the same message

**Comment by jtesta** [comment]
Ahh... I see the problem now.  Your `D:\hg\Rainbow Crackalack v1.0\rt_ntlm` directory has RTI2 tables, not RT/RTC tables.

The Rainbow Crackalack lookup tool only supports the RT and RTC formats.  Its not finding any of those files in the directory you gave it, so it can't continue.  But if you point it to the NTLM 8 tables (available from [the project website](https://www.rainbowcrackalack.com/#download)), it'll work.

Technically, you could use the `rcracki_mt` tool [from this project](https://sourceforge.net/projects/rcracki/) with those RTI2 tables that you have, but honestly... those are pretty much out-dated at this point.  You'd be doing a lot of work just to find passwords with just lowercase characters, numbers, and spaces (no upper case nor special chars!).

Secondly, I notice that you're trying to crack almost 45,000 NTLM hashes, which is a HUGE amount.  Cracking with rainbow tables first requires a pre-computation step done upon each hash.  My 2 x Vega 64 machine takes 16.5 seconds per hash.  Hence, the pre-computation steps alone would take 44,881 x 16.5 = 740537 seconds = 206 hours!  In contrast, simply running hashcat on the same machine to brute-force all 8-character passwords would take just 38 hours.

The bottom line here is that rainbow tables are only efficient to use for smaller numbers of hashes.  Assuming that 45K is a dump of hashes from a Windows domain, try targeting just the domain admins and other high-privilege accounts.

**Comment by jtesta** [comment]
I believe my last comment solved the problem, so I'm closing this.  If you run into other problems, please let me know!
